src/commonMain/kotlin/com/charleskorn/kaml/YamlOutput.kt

```diff
@@ -81,16 +81,25 @@ internal class YamlOutput(
     override fun encodeLong(value: Long) = emitPlainScalar(value.toString())
     override fun encodeShort(value: Short) = emitPlainScalar(value.toString())
 
-    private var forcedSingleLineScalarStyle: ScalarStyle? = null
-    private var forcedMultiLineScalarStyle: ScalarStyle? = null
+    private var forcedSingleLineScalarStyle: StringScalarStyle? = null
+    private var forcedMultiLineScalarStyle:  StringScalarStyle? = null
+
+    private fun StringScalarStyle.toScalarStyle() =
```

Comment:
suggestion: I am not sure if it is needed at all but it would be probably good to know if there is a `ScalarStyle` value that is not covered by the `StringScalarStyle` enum (in case the new value is added in one of the versions).

To do that we probably should create a test that iterates over `StringScalarStyle`'s values, converts them to `ScalarStyle` and compares the resulted set with all values available in `ScalarStyle` and reports the difference

Reply:
If that happens the code will fail to build:
![image](https://github.com/user-attachments/assets/79928a73-fc82-477d-bde0-9d01d817bfe9)

So no test is needed to ensure this.

Reply:
I double checked this is not only an IDEA error; also the compile fails over this.

Reply:
Hi, @nielsbasjes. This compile error only happens if you have a new value added to `StringScalarStyle`. However, if a new value is added to `ScalarStyle` the code will still compile (because the `when` expression will still be exhausted). As I said, I am not sure if this test is required (YAML spec does not have other types of scalar as far as I know) but still it could be good to have a test to cover this.

Also, nobody is safe from a bug like this (but this is another problem...):
```kotlin
return when (this) {
    StringScalarStyle.DOUBLE_QUOTED -> ScalarStyle.DOUBLE_QUOTED
    StringScalarStyle.SINGLE_QUOTED -> ScalarStyle.SINGLE_QUOTED
    StringScalarStyle.LITERAL       -> ScalarStyle.LITERAL
    StringScalarStyle.FOLDED        -> ScalarStyle.LITERAL // duplicate
    StringScalarStyle.PLAIN         -> ScalarStyle.PLAIN
}
```

Reply:
Test added.